presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoControlUtils.java

```diff
@@ -0,0 +1,197 @@
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static com.teradata.prestomanager.agent.AgentFileUtils.getFileProperty;
+import static com.teradata.prestomanager.agent.CommandExecutor.executeCommand;
+import static com.teradata.prestomanager.agent.PrestoControlUtils.StopType.GRACEFUL;
+import static com.teradata.prestomanager.common.RequestBuilder.newRequestBuilder;
+import static java.lang.String.format;
+import static javax.ws.rs.client.Entity.entity;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON_TYPE;
+import static javax.ws.rs.core.Response.Status.ACCEPTED;
+import static javax.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR;
+import static javax.ws.rs.core.Response.Status.NOT_FOUND;
+import static javax.ws.rs.core.Response.Status.OK;
+
+public final class PrestoControlUtils
+{
+    private static final Logger LOGGER = LogManager.getLogger(PrestoControlUtils.class);
+    private static final String LAUNCHER_SCRIPT = "/usr/lib/presto/bin/launcher";
+    private static final Path PRESTO_CONFIG_FILE = Paths.get("/etc/presto/config.properties");
+    private static final Gson GSON = new Gson();
+    private static final int SUBPROCESS_TIMEOUT = 120;
+
+    private PrestoControlUtils() {}
+
+    public static Response startUsingRpm()
+    {
+        try {
+            if (!isInstalled()) {
+                return Response.status(NOT_FOUND).entity("Presto is not installed").build();
+            }
+        }
+        catch (PrestoManagerException e) {
+            LOGGER.error(e.getMessage(), e.getCause());
+            return Response.status(INTERNAL_SERVER_ERROR).build();
+        }
+        new Thread(() -> {
+            try {
+                int startPresto = executeCommand(SUBPROCESS_TIMEOUT, "service", "presto", "start");
+                if (startPresto != 0) {
+                    LOGGER.error("Failed to start presto");
+                }
+            }
+            catch (PrestoManagerException e) {
+                LOGGER.error(e.getMessage(), e.getCause());
+            }
+        }).start();
+        return Response.status(ACCEPTED).entity("Presto is being started. Please check back later using status.").build();
+    }
+
+    public static Response stopUsingRpm(StopType stopType)
+    {
+        try {
+            if (!isInstalled()) {
+                return Response.status(NOT_FOUND).build();
+            }
+            if (!isRunning()) {
+                return Response.status(OK).build();
+            }
+            stopHelper(stopType);
+        }
+        catch (PrestoManagerException e) {
+            LOGGER.error(e.getMessage(), e.getCause());
+            return Response.status(INTERNAL_SERVER_ERROR).build();
+        }
+        return Response.status(OK).build();
+    }
+
+    private static void stopHelper(StopType stopType)
+            throws PrestoManagerException
+    {
+        switch (stopType) {
```

Comment:
it might be clearer if you put the switch statement in stopUsingRpm and had separate methods for each case.